.changelog/225.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+fix SPIFFE handling for Vault connect certificates
```

Comment:
I'm fine with making it more generic, this will affect any custom cert configuration you can have for the connect CA, whether Vault or the [bring-your-own option](https://www.consul.io/docs/connect/ca/consul#specifying-a-custom-private-key-and-root-certificate).